src/unit_tests/misc/library_test.cpp

```diff
@@ -0,0 +1,195 @@
+
+        CHECK(lib.get_types().size()             == 0);
+        CHECK(lib.get_global_properties().size() == 0);
+        CHECK(lib.get_global_methods().size()    == 0);
+        CHECK(does_plugin_type_exist()           == false);
+    }
+
+    SECTION("negative test")
+    {
+        library lib("SomeNonExisitingLibrary");
+
+        CHECK(lib.is_loaded()                    == false);
+        CHECK(lib.get_types().size()             == 0);
+        CHECK(lib.get_global_properties().size() == 0);
+        CHECK(lib.get_global_methods().size()    == 0);
+        CHECK(lib.get_error_string().empty()     == true);
+
+        CHECK(lib.load()                         == false);
+
+        CHECK(lib.is_loaded()                    == false);
+        CHECK(lib.get_error_string().empty()     == false);
+        CHECK(lib.unload()                       == false);
+    }
+
+    SECTION("load with suffix")
+    {
+#if RTTR_PLATFORM == RTTR_PLATFORM_WINDOWS
+        library lib(library_name + std::string(".dll"));
+#elif RTTR_PLATFORM == RTTR_PLATFORM_LINUX
+        library lib(library_name + std::string(".so"));
+#elif RTTR_PLATFORM == RTTR_PLATFORM_APPLE
+        library lib(library_name + std::string(".dylib"));
+#else
+    #error "Don't know library suffix on this plattform!"
+#endif
+        CHECK(does_plugin_type_exist()          == false);
+        CHECK(lib.load()                        == true);
+        REQUIRE(lib.get_error_string().empty()  == true);
+        CHECK(does_plugin_type_exist()          == true);
+
+        CHECK(lib.get_error_string().empty()    == true);
+        CHECK(lib.unload()                      == true);
+
+        CHECK(does_plugin_type_exist()          == false);
+    }
+}
+
+/////////////////////////////////////////////////////////////////////////////////////////
+
+TEST_CASE("library - multi load", "[library]")
+{
+    SECTION("load multiple times")
+    {
+        library lib(library_name);
+
+        CHECK(does_plugin_type_exist()              == false);
+        CHECK(lib.load()                            == true);
+        CHECK(lib.load()                            == true);
+
+        REQUIRE(lib.get_error_string().empty()      == true);
+        CHECK(does_plugin_type_exist()              == true);
+
+        CHECK(lib.get_types().size()                > 1);
+        CHECK(lib.get_global_properties().size()    == 0);
+        CHECK(lib.get_global_methods().size()       == 0);
+
+        CHECK(lib.unload() == true);
+        CHECK(does_plugin_type_exist() == false);
+    }
+
+    SECTION("load the same lib multiple times")
+    {
+        library lib1(library_name);
+        library lib2(library_name);
+
+        CHECK(does_plugin_type_exist() == false);
+
+        CHECK(lib1.load() == true);
+        CHECK(lib2.load() == true);
+
+        CHECK(does_plugin_type_exist() == true);
+
+        CHECK(lib1.unload() == false);
+        CHECK(lib1.unload() == false);
```

Comment:
another unload -> false because there is still lib2 there

Reply:
The second unload does nothing, it was still unloaded.
And yes lib2 is still loaded, so no unload at all possible.